sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
             else if (dest_len == 2 &&
                      ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
                      ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
+                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
```

Comment:
My personal opinion is with @authmillenon on this issue, the code is less readable if you are using `.uint16` and it will look like you are doing double byte order swaps after adding a HTONS to the mix. The byte swapping will result in the same or similar machine code as the two `.uint8` assignments.